app/src/main/java/mozilla/lockbox/store/PublicPreferencesStore.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.lockbox.store
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.preference.PreferenceManager
+import io.reactivex.Observable
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.addTo
+import mozilla.lockbox.R
+import mozilla.lockbox.action.ItemListSort
+import mozilla.lockbox.action.SettingsAction
+import mozilla.lockbox.extensions.filterByType
+import mozilla.lockbox.flux.Dispatcher
+
+open class PublicPreferencesStore(val dispatcher: Dispatcher = Dispatcher.shared) {
+
+    internal val compositeDisposable = CompositeDisposable()
+    internal lateinit var sharedPrefs: SharedPreferences
+
+    companion object {
+        val shared = PublicPreferencesStore()
+    }
+
+    private lateinit var itemListSortKey: String
+    val itemListSortObservable: Observable<ItemListSort> by lazy {
+        val onNext: (String, Int) -> ItemListSort = { key, default ->
+            ItemListSort.fromSortId(sharedPrefs.getInt(key,default))!!
+        }
+        createObservableForKey(itemListSortKey, ItemListSort.ALPHABETICALLY.sortId, onNext)
+    }
+
+    fun applyContext(context: Context) {
+        sharedPrefs = PreferenceManager.getDefaultSharedPreferences(context)
+
+        itemListSortKey = context.resources.getString(R.string.shared_prefs_sort_option_key)
```

Comment:
Does this need to be in resources, as it's an opaque string key?

Reply:
It doesn't need to be. I guess it would depend on whether we are likely to use this key in more than one place. Seeing as at this time it is only used here, I shall move it to a string literal - it can always be moved back to resources if we need to reference it in multiple places.